homework-submission/week2hw/src/App.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from "react";
+import "./App.css";
+import todoList from "./data/todos.json";
+import Description from "./components/Description";
+
+class App extends Component {
+  state = {
+    todoList
+  };
+
+  clickHandler = e => {
+    let tempArr = this.state.todoList.slice();
+
+    console.log(e.target.id);
+
+    tempArr[e.target.id - 1].done = !tempArr[e.target.id - 1].done;
+
+    this.setState({
+      todoList: tempArr
+    });
+  };
+
+  render() {
+    let renderList;
+    if (todoList.length) {
+      renderList = todoList.map(todo => {
+        return (
+          <div key={todo.id}>
+            <input
+              type="checkbox"
+              defaultChecked={todo.done}
+              id={todo.id}
+              onClick={e => this.clickHandler(e)}
```

Comment:
If you are going to call `clickHandler` with sending the `e` directly, then you can write it like this `onClick={this.clickHandler}` because this equals what you are doing exactly.

Reply:
Thanks, i didn't know that